src/macros.jl

```diff
@@ -317,9 +317,18 @@ macro grad_from_chainrules(fcall)
     Meta.isexpr(fcall, :call) && length(fcall.args) >= 2 ||
         error("`@grad_from_chainrules` has to be applied to a function signature")
     f = esc(fcall.args[1])
-    xs = fcall.args[2:end]
+    xs = map(fcall.args[2:end]) do x
+        if x isa Expr && x.head == :(::)
+            if length(x.args) == 1 # ::T without var name
+                return :($(gensym())::$(esc(x.args[1])))
+            else # x::T
```

Comment:
Would be safer to check that `length(x.args) == 2` and error otherwise.

Reply:
Do you know of a case where head is `:(::)` and args are more than 2? It's a syntax error in Julia now if I try to construct the expression manually and eval it.

Reply:
if it's a syntax error, I can't test the third else branch